app/models/lecture.js

```diff
@@ -18,35 +18,38 @@ module.exports = (sequelize, DataTypes) => {
         //         model: Course,
         //         key: 'id'
         //     },
-        //     unique: 'course_order_index' // compound unique constraint with 'order'
+        //     unique: {    // compound unique constraint with 'courseId'
+        //         args: 'course_order_index',
+        //         msg: 'There already exists a lecture with this order number in this course'
+        //     }        
         // },
         title: {
             type: DataTypes.STRING(50), // max length of 50
             allowNull: false,
             validate: {
                 notNull: {
-                    msg: "Lecture name required"
+                    msg: 'Lecture name required'
                 },
-                notEmpty: {
-                    msg: "Lecture name cannot be empty"
-                },
-                max: {
-                    args: [50],
-                    msg: "Title must be less than 50 characters"
+                len: {
+                    args: [1, 50],
+                    msg: 'Title must be between 1 and 50 characters'
                 }
             }
         },
         // the order of current lecture within a course
         order: {
```

Comment:
I don't think it's super critical, but it might make sense to add a custom hook beforeCreate that sets the order to the next logical number if a value is not passed in